pallets/nfts/src/features/transfer.rs

```diff
@@ -87,16 +89,21 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		with_details(&collection_details, &mut details)?;
 
 		// Update account balance of the owner.
-		let balance = AccountBalance::<T, I>::take(collection, &details.owner)
-			.checked_sub(1)
-			.ok_or(Error::<T, I>::NoItemOwned)?;
-		if balance > 0 {
-			AccountBalance::<T, I>::insert(collection, &details.owner, balance);
-		}
+		Self::decrement_account_balance(collection, &details.owner)?;
+
 		// Update account balance of the destination account.
-		AccountBalance::<T, I>::mutate(collection, &dest, |balance| {
-			balance.saturating_inc();
-		});
+		let deposit_amount =
+			match collection_config.is_setting_enabled(CollectionSetting::DepositRequired) {
+				true => T::BalanceDeposit::get(),
+				false => Zero::zero(),
+			};
+		// If the destination account exists, it reserves the `BalanceDeposit` for the new
+		// `AccountBalance`. Otherwise, the caller is responsible.
+		let deposit_account = match frame_system::Pallet::<T>::account_exists(&dest) {
```

Comment:
```suggestion
		let deposit_account = match T::Currency::free_balance(&dest) {
```

Curious if checking the balance makes more sense than checking the account existence?